### ISSUE TITLE
Error with templates exporting config as function

### ISSUE THREAD

**Issue by lejoe** [saojs/sao#77]
First of all, thanks a lot for this great tool!
I love the simplicity!

We have a (private) template that [exports config as function like described in the doc](https://sao.js.org/#/create?id=exporting-a-function). 

When running `sao org/template-repo . -u --clone` we get this error:
```
error TypeError: getConfigOptions is not a function
    at fs.pathExists.then.exists (/usr/local/lib/node_modules/sao/lib/utils/config.js:38:16)
    at <anonymous>
```

I've looked at the code a bit, and I might be related with this change: https://github.com/saojs/sao/commit/425f8cffb9c24a37f51c3a9d6fb413357aa63efb

It looks like from `index.js` you call a function from `config.js` which itself requires a function from `index.js`.

TL;DR: here is an example of what your code does and produces the same error: https://repl.it/repls/GranularWeakTitles
